### PR TITLE
1399 order of household register

### PR DESCRIPTION
**Pull request by AbdulWahabMemon**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #https://github.com/onaio/fhir-resources/issues/1399

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

**Review by ageryck — DISMISSED** [review]

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt [hunk 1]
```diff
@@ -659,8 +659,8 @@ constructor(
       )
     ) {
       group.addMember(Group.GroupMemberComponent().apply { entity = reference })
-      defaultRepository.addOrUpdate(resource = group)
     }
+    defaultRepository.addOrUpdate(resource = group)
```
[the remark this answers is not in the document]
Comment (ageryck) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt:663]: The group resource is a base resource on the register/profile, we want to indicate it as updated as long as related resources are. The pros to this it allows us to filter the group with the latest interaction to the register top but the con is that we will interfere with the audit trail of group resource-specific edits

[the remark this answers is not in the document]
Comment (dubdabasoduba) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt:663]: @ageryck I think the above use case is already handled by configs. @SebaMutuku worked on something that allowed us to define the group to update once a resource tied to the group is altered

[the remark this answers is not in the document]
Comment (dubdabasoduba) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt:663]: @AbdulWahabMemon In the cases where the group is not updated to we have it defined on the questionnaire configs that open the forms that create the said resources

**Review by Rkareko — APPROVED** [review]

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (dubdabasoduba) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt:663]: NVM my question above I checked the code. The change makes sense

[the remark this answers is not in the document]
Comment (Rkareko) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt:663]: @ageryck @AbdulWahabMemon Could you add a short comment on the code that explains the need for the change

[the remark this answers is not in the document]
Comment (AbdulWahabMemon) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt:663]: > @AbdulWahabMemon In the cases where the group is not updated to we have it defined on the questionnaire configs that open the forms that create the said resources?

for every questionnaire with groupIdentifier, group is updated with _lastUpdated but only store in db when passing the check at line 662 in QuestionnaireViewModel, 
that fails to sorting group in register if there is interaction inside group like Tasks, Encounter, Observations, are created/updated for a group, these ResourceType check is not at that line, so either we can add those ResourceType check or we can update group outside that check.

[the remark this answers is not in the document]
Comment (AbdulWahabMemon) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt:666]: @Rkareko comment added

**Review by ageryck — DISMISSED** [review]

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt [hunk 2]
```diff
@@ -659,8 +659,11 @@ constructor(
       )
     ) {
       group.addMember(Group.GroupMemberComponent().apply { entity = reference })
-      defaultRepository.addOrUpdate(resource = group)
     }
+    // here Group is coming with groupIdentifier thus it make sense that
+    // there is an interaction inside group rather addingMemberItems only for above check
+    // so for every cases group is to be updated
```
Comment (dubdabasoduba) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt:665]:
```suggestion
    // here Group is coming with groupIdentifier thus it make sense that
    // there is an interaction inside group rather addingMemberItems only for above check
    // so for every cases group is to be updated
    
    /**
    * The Group Resource is fetched by the `groupIdentifier`.  
    * We trigger the group update every time anything linked to it in order to change the `_lastUpdated` timestamp.
    * This helps us with order the last updated group (household) on the top of the register. 
    */
```

**Review by dubdabasoduba — CHANGES_REQUESTED** [review]

**Review by dubdabasoduba — APPROVED** [review]

**Review by dubdabasoduba — APPROVED** [review]

**Pull request merged**